### PR TITLE
Add DistSQL UNLOCK CLUSTER

### PR DESCRIPTION
**Pull request by Pace2Car**
For #24338.

Changes proposed in this pull request:
  - Add DistSQL UNLOCK CLUSTER
  - Add related test cases

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/updatable/UnlockClusterUpdater.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import org.apache.shardingsphere.mode.lock.GlobalLockDefinition;
+import org.apache.shardingsphere.mode.manager.ContextManager;
+import org.apache.shardingsphere.mode.manager.cluster.coordinator.registry.status.cluster.event.ClusterStatusChangedEvent;
+import org.apache.shardingsphere.proxy.backend.context.ProxyContext;
+
+/**
+ * Unlock cluster updater.
+ */
+@RequiredArgsConstructor
+@Setter
```
Comment (RaigorJiang) [inline proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/updatable/UnlockClusterUpdater.java:37]: Please remove this @setter.

File: proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/updatable/UnlockClusterUpdater.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    
+    private void checkMode() {
+        ShardingSpherePreconditions.checkState(ProxyContext.getInstance().getContextManager().getInstanceContext().isCluster(),
+                () -> new UnsupportedSQLOperationException("Only allowed in cluster mode"));
+    }
+    
+    private void checkState() {
+        ClusterState currentState = ProxyContext.getInstance().getContextManager().getClusterStateContext().getCurrentState();
+        ShardingSpherePreconditions.checkState(ClusterState.OK != currentState, () -> new IllegalStateException("Cluster does not locked"));
```
Comment (RaigorJiang) [inline proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/updatable/UnlockClusterUpdater.java:59]: `is` not locked.

File: proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/updatable/UnlockClusterUpdater.java (hunk 1 above)
Comment (Pace2Car) [reply proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/updatable/UnlockClusterUpdater.java]: THX, I have Fixed.

File: proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/updatable/UnlockClusterUpdater.java (hunk 2 above)
Comment (Pace2Car) [reply proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/updatable/UnlockClusterUpdater.java]: THX, I have Fixed.

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**